lib/decision/bucketing_manager.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:flagship/api/endpoints.dart';
+import 'package:flagship/api/service.dart';
+import 'package:flagship/decision/bucketing_process.dart';
+import 'package:flagship/decision/decision_manager.dart';
+import 'package:flagship/decision/polling/polling.dart';
+import 'package:flagship/flagship.dart';
+import 'package:flagship/flagship_config.dart';
+import 'package:flagship/flagship_version.dart';
+import 'package:flagship/model/bucketing.dart';
+import 'package:flagship/model/campaigns.dart';
+import 'package:flagship/utils/logger/log_manager.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class BucketingManager extends DecisionManager {
+  final int intervalPolling;
+  Polling? polling;
+  bool fileExists = true;
+
+  late Future<SharedPreferences> _prefs = SharedPreferences.getInstance();
+  late Campaigns campaigns;
+
+  String lastModfiedKey = "FSLastModifiedScript";
+  String bucketingFolder = "/flagship/Bucketing/";
+  String fileName = "bucketing.json";
+
+  BucketingManager(Service service, this.intervalPolling) : super(service) {
+    // startPolling();
+  }
+
+  @override
+  Future<Campaigns> getCampaigns(
+      String envId, String visitorId, String? anonymousId, Map<String, Object> context) async {
+    // Read File before
+    String jsonString = await _readFile().catchError((error) {
+      Flagship.logger(Level.ALL, "Error on reading the saved bucketing or the file doesn't exist");
+      throw Exception('Flagship, Failed to synchronize');
```

Comment:
The file will not exist on the first call to getCampaigns, no? So the method will always throw and exception ? :thinking:

Reply:
Yes you're right, for user throwing and exception on get campaign at the first start give an uncomfortable situation , so will return an empty response with a log that say "**Flagship, Failed to synchronize**"